docs/main/create-ocp-project/README.md

```diff
@@ -14,97 +14,176 @@ The following table lists the configurable parameters of the Create OCP project
 
-To do so, edit the secret `${WORKFLOW_NAME}-creds` and set those values and the one of `NOTIFICATIONS_BEARER_TOKEN`:
-```
-WORKFLOW_NAME=create-ocp-project
-oc -n sonataflow-infra patch secret "${WORKFLOW_NAME}-creds" --type merge -p '{
-   "data":{
-      "NOTIFICATIONS_BEARER_TOKEN":"'$(oc get secrets -n rhdh-operator backstage-backend-auth-secret -o go-template='{{ .data.BACKEND_SECRET  }}')'"
-   },
-   "stringData":{
-      "JIRA_API_TOKEN":"{{ JIRA_API_TOKEN }}",
-      "OCP_API_SERVER_TOKEN":"{{ OCP_API_SERVER_TOKEN }}"
-   }
-}'
+To obtain an OpenShift API token, create a Service Account, assign permissions to it, and request a token:
+
+```bash
+oc create sa orchestrator-ocp-api
+oc adm policy add-cluster-role-to-user cluster-admin -z orchestrator-ocp-api
+oc create token orchestrator-ocp-api
 ```
-If you are using Jira cloud, you can generate the `JIRA_API_TOKEN` using https://id.atlassian.com/manage-profile/security/api-tokens 
 
-The `OCP_API_SERVER_TOKEN` should be associated with a service account.
+### Add the Environment Variables to a Secret
+
+The Helm Chart installation creates a `create-ocp-project-creds` Secret in the
+namespace where it's installed. You'll update this Secret with your environment
+variable values.
+
+> [!NOTE]
+> Updating the Secret does not automatically restart the workflow Pod, nor update the associated Sonataflow CR. If you update the keys in the secret you must update the Sonataflow CR - as shown below - to reference them. If you update the values in the Secret you must delete the existing `create-ocp-project` workflow Pod to create a new Pod that uses the new values.
+
+> [!WARNING]
+> If you run the `helm upgrade` command, the values of the Secret are reset.
+
+Run the following command to do so. Replace the example values with the
+correct values for your environment:
+
+```bash
+export TARGET_NS='sonataflow-infra'
+export WORKFLOW_NAME='create-ocp-project'
+
+export NOTIFICATIONS_BEARER_TOKEN='token_from_backstage_appconfig'
+export BACKSTAGE_NOTIFICATIONS_URL='https://backstage.cluster.com'
+
+export JIRA_API_TOKEN='token_for_jira_api'
+export JIRA_URL='https://foo-bar.atlassian.net/'
+export JIRA_USERNAME='foo@bar.com'
+
+export OCP_API_SERVER_URL='https://api.cluster.replaceme.com:6443'
+export OCP_API_SERVER_TOKEN='replaceme'
+export OCP_CONSOLE_URL='replaceme'
+
+oc -n $TARGET_NS patch secret "$WORKFLOW_NAME-creds" \
+  --type merge -p "{ \
+    \"stringData\": { \
+      \"NOTIFICATIONS_BEARER_TOKEN\": \"$NOTIFICATIONS_BEARER_TOKEN\" \
+      \"JIRA_API_TOKEN\": \"$JIRA_API_TOKEN\", \
+      \"OCP_API_SERVER_TOKEN\": \"$OCP_API_SERVER_TOKEN\", \
+      \"BACKSTAGE_NOTIFICATIONS_URL\": \"$BACKSTAGE_NOTIFICATIONS_URL\", \
+      \"JIRA_URL\": \"$JIRA_URL\", \
+      \"JIRA_USERNAME\": \"$JIRA_USERNAME\", \
+      \"OCP_API_SERVER_URL\": \"$OCP_API_SERVER_URL\", \
+      \"OCP_CONSOLE_URL\": \"$OCP_CONSOLE_URL\" \
+    } \
+  }"
```

Comment:
This should be
```
oc -n $TARGET_NS patch secret "$WORKFLOW_NAME-creds" \
  --type merge -p "{
    \"data\": {
        \"NOTIFICATIONS_BEARER_TOKEN\": \"$NOTIFICATIONS_BEARER_TOKEN\"
    },
    \"stringData\": { 
      \"JIRA_API_TOKEN\": \"$JIRA_API_TOKEN\", 
      \"OCP_API_SERVER_TOKEN\": \"$OCP_API_SERVER_TOKEN\", 
      \"BACKSTAGE_NOTIFICATIONS_URL\": \"$BACKSTAGE_NOTIFICATIONS_URL\", 
      \"JIRA_URL\": \"$JIRA_URL\", 
      \"JIRA_USERNAME\": \"$JIRA_USERNAME\", 
      \"OCP_API_SERVER_URL\": \"$OCP_API_SERVER_URL\", 
      \"OCP_CONSOLE_URL\": \"$OCP_CONSOLE_URL\" 
    }
  }"
```

Reply:
This assumes the Backstage instance is deployed with the Orchestrator Operator. I think the instructions should be more generic, so if someone has deployed using the RHDH Operator/Helm chart we can tell them where to look. WDYT/

Reply:
For example, there's no `backstage-backend-auth-secret` if you're using the official operator. Instead this value is set like so:

![Screenshot 2024-11-04 at 10 46 38 AM](https://github.com/user-attachments/assets/50da7433-8036-4693-9a09-c27412f71b39)

Reply:
> This assumes the Backstage instance is deployed with the Orchestrator Operator. I think the instructions should be more generic, so if someone has deployed using the RHDH Operator/Helm chart

That's a valid point: currently we take as assumption that the RHDH was installed using the Orchestrator Operator, @pkliczewski should we keep doing that?

> we can tell them where to look.

Yes, we can have 2 section: 

1. RHDH installed with RHDH operator: in this section we have 
```
export NOTIFICATIONS_BEARER_TOKEN=$(oc get secrets -n rhdh-operator backstage-backend-auth-secret -o go-template='{{ .data.BACKEND_SECRET  }}' | base64 -d)
```
Not the ` | base64 -d` , this way all values will be going to `stringData` in the secret's patch
2. RHDH not installed with RHDH operator: in this section we have instructions on where to look to get the token and then
```
export NOTIFICATIONS_BEARER_TOKEN='replace_me'
```

The rest should not change:
```
oc -n $TARGET_NS patch secret "$WORKFLOW_NAME-creds" \
  --type merge -p "{
    \"stringData\": { 
       \"NOTIFICATIONS_BEARER_TOKEN\": \"$NOTIFICATIONS_BEARER_TOKEN\",
      \"JIRA_API_TOKEN\": \"$JIRA_API_TOKEN\", 
      \"OCP_API_SERVER_TOKEN\": \"$OCP_API_SERVER_TOKEN\", 
      \"BACKSTAGE_NOTIFICATIONS_URL\": \"$BACKSTAGE_NOTIFICATIONS_URL\", 
      \"JIRA_URL\": \"$JIRA_URL\", 
      \"JIRA_USERNAME\": \"$JIRA_USERNAME\", 
      \"OCP_API_SERVER_URL\": \"$OCP_API_SERVER_URL\", 
      \"OCP_CONSOLE_URL\": \"$OCP_CONSOLE_URL\" 
    }
  }"
```

WDYT?

Reply:
Sounds good 👍